docs/website/docs/command-reference/deploy.md

```diff
@@ -0,0 +1,66 @@
+---
+title: odo deploy
+sidebar_position: 4
+---
+
+# odo deploy
+
+odo can be used to deploy components in a similar manner they would be deployed by a CI/CD system, 
+by first building the images of the containers to deploy, then by deploying the Kubernetes resources
+necessary to deploy the components.
+
+When running the command `odo deploy`, odo searches for the default command of kind `deploy` in the devfile, and executes this command.
+The `deploy` command is typically a *composite* command, composed of several *apply* commands:
+- a command referencing an `Image` component that, when applied, will build the image of the container to deploy,
+- a command referencing a `Kubernetes` component that, when applied, will create Kubernetes resources in the cluster.
+
+With the following example `devfile.yaml` file, a container image will be built by using the `Dockerfile` present in the directory,
```

Comment:
I would prefer not to add a complete devfile. There are a lot of mandatory parts in the devfile that won't be related to this subject, that ill make a very long file and will be difficult for the reader to focus on the important parts
I added the `schemaVersion` to highlight that it should be >= 2.2.0

Reply:
OK. Makes sense. If you think adding a link to a fully functional devfile would make sense, then you could add it. Otherwise, it's OK. We can revisit the discussion if anyone asks for such a devfile. 😄